### PR TITLE
[cleanup] Consolidate some query and controller/broker methods in integration tests

### PR DESCRIPTION
**Pull request by xiangfu0**
- Remove unnecessary usage of static `PostQuery` calls.
- Remove direct reference to `_controllerBaseApiUrl`, `_controllerPort` and `_brokerBaseApiUrl` in extended test classes.
- Always add `useMultistageEngine` query option if possible.
- Simplify MultiStageEngineCustomTenantIntegrationTest

### CODE REVIEW

File: pinot-controller/src/test/java/org/apache/pinot/controller/helix/ControllerTest.java [hunk 1]
```diff
@@ -119,8 +119,8 @@ public class ControllerTest {
 
   protected static HttpClient _httpClient = null;
 
-  protected int _controllerPort;
-  protected String _controllerBaseApiUrl;
```
Comment (Jackie-Jiang) [inline pinot-controller/src/test/java/org/apache/pinot/controller/helix/ControllerTest.java:123]: Any specific reason to hide them? IMO it is okay to directly access them

**Review by Jackie-Jiang — COMMENTED** [review]
Please take a look at the failed test

File: pinot-controller/src/test/java/org/apache/pinot/controller/helix/ControllerTest.java (hunk 1 above)
Comment (xiangfu0) [reply pinot-controller/src/test/java/org/apache/pinot/controller/helix/ControllerTest.java:123]: This is more to simplify the usage.
Eventually all the usage of `_controllerBaseApiUrl` should go to `getControllerRequestClient()` or `getControllerRequestURLBuilder`, as well as `_brokerBaseApiUrl`.

**Comment by xiangfu0** [comment]
> Please take a look at the failed test

Fixed.

**Review by Jackie-Jiang — APPROVED** [review]

**Pull request merged**